tests/devices/test_null_qubit.py

```diff
@@ -11,1377 +11,1241 @@
+        params = qml.numpy.array(rng.random(shape))
+        dev = qml.device("null.qubit")
 
-class TestTrackerIntegration:
-    """Tests tracker integration behavior with 'null.qubit'."""
+        diff_method = "device"
 
-    def test_single_execution(self, nullqubit_device, mocker):
-        """Test correct behavior with single circuit execution"""
-        dev = nullqubit_device(wires=1)
+        @qml.qnode(dev, diff_method=diff_method)
+        def circuit(params):
+            qml.StronglyEntanglingLayers(params, wires=range(n_wires))
+            return [qml.expval(qml.Z(i)) for i in range(n_wires)]
 
-        @qml.qnode(dev)
-        def circuit():
-            qml.RX(1.23, wires=0)
-            qml.CNOT(wires=[0, 1])
-            return qml.expval(qml.PauliZ(0))
+        def cost(params):
+            out = circuit(params)
+            return qml.math.sum(out)
 
-        circuit_resources = qml.resource.Resources(
-            num_wires=2,
-            num_gates=2,
-            gate_types={"RX": 1, "CNOT": 1},
-            gate_sizes={1: 1, 2: 1},
-            depth=2,
-        )
+        assert np.array_equal(qml.grad(cost)(params), np.zeros(shape))
 
-        class callback_wrapper:
-            """Callback wrapping class"""
-
-            # pylint: disable=too-few-public-methods
-
-            @staticmethod
-            def callback(totals=None, history=None, latest=None):
-                """Wrapped callback function."""
-
-        wrapper = callback_wrapper()
-        spy = mocker.spy(wrapper, "callback")
-
-        with Tracker(circuit.device, callback=wrapper.callback) as tracker:
-            circuit()
-            circuit()
-
-        assert tracker.totals == {"executions": 2, "batches": 2, "batch_len": 2}
-        assert tracker.history == {
-            "executions": [1, 1],
-            "shots": [None, None],
-            "batches": [1, 1],
-            "batch_len": [1, 1],
-            "resources": [circuit_resources, circuit_resources],
-        }
-        assert tracker.latest == {"batches": 1, "batch_len": 1}
-
-        _, kwargs_called = spy.call_args_list[-1]
-
-        assert kwargs_called["totals"] == {"executions": 2, "batches": 2, "batch_len": 2}
-        assert kwargs_called["history"] == {
-            "executions": [1, 1],
-            "shots": [None, None],
-            "batches": [1, 1],
-            "batch_len": [1, 1],
-            "resources": [circuit_resources, circuit_resources],
-        }
-        assert kwargs_called["latest"] == {"batches": 1, "batch_len": 1}
-
-    def test_shots_execution(self, nullqubit_device, mocker):
-        """Test that correct tracks shots."""
-        # pylint: disable=unexpected-keyword-arg
-        dev = nullqubit_device(wires=1)
 
-        @qml.qnode(dev)
-        def circuit():
-            qml.RX(1.23, wires=0)
-            qml.CNOT(wires=[0, 1])
-            return qml.expval(qml.PauliZ(0))
+@pytest.mark.parametrize("method", ["device", "parameter-shift"])
```

Comment:
Sorry... one more case.  Can we test with both `diff_method="device", device_vjp=True` and `diff_method="device", device_vjp=False`?

Reply:
done, except.. jax.jacobian didn't love it